src/nnbench/runner.py

```diff
@@ -263,6 +264,7 @@ def run(
                 "date": datetime.now().isoformat(timespec="seconds"),
                 "error_occurred": False,
                 "error_message": "",
+                "parameters": {**bmdefaults, **bmparams},
```

Comment:
Does this ensure that the defaults are overwritten by the `bmparams`?

I think something like `bmdefaults.update(bmparams)` would be easier to read.

Reply:
I think this does what it needs to do: For any variable, if the default is used, it's not in `bmparams`, so the update does nothing - and for every overridden default, the `update()` overwrites with the argument given in the params dict.

So the order should be defaults, then update with params - which is what I suggested (you can also write `bmdefaults | bmparams`).

Reply:
If you need a mental model, it helps to think about what happens once the `params` are passed into the function - its default keyword arguments are already there, and everything else gets overridden by the incoming parameters. So it is indeed `bmdefault.update(bmparams)` we're looking for here (you can even use that as the dict value, since we do not explicitly use it again afterwards).